src/dbnode/storage/index/block_test.go

```diff
@@ -702,7 +702,107 @@ func TestBlockMockQueryLimitExhaustive(t *testing.T) {
 	ctx := context.NewContext()
 
 	exhaustive, err := b.Query(ctx, resource.NewCancellableLifetime(),
-		defaultQuery, QueryOptions{Limit: limit}, results, emptyLogFields)
+		defaultQuery, QueryOptions{SeriesLimit: limit}, results, emptyLogFields)
+	require.NoError(t, err)
+	require.True(t, exhaustive)
+
+	rMap := results.Map()
+	require.Equal(t, 1, rMap.Len())
+	t1, ok := rMap.Get(ident.StringID(string(testDoc1().ID)))
+	require.True(t, ok)
+	require.True(t, ident.NewTagIterMatcher(
+		ident.MustNewTagStringsIterator("bar", "baz")).Matches(
+		t1))
+
+	// NB(r): Make sure to call finalizers blockingly (to finish
+	// the expected close calls)
+	ctx.BlockingClose()
+}
+
+func TestBlockMockQueryDocsLimit(t *testing.T) {
```

Comment:
Maybe rename this to `TestBlockMockQueryDocsLimitNonExhaustive` (to match the opposite `TestBlockMockQueryDocsLimitExhaustive`)?